File: src/main/java/dev/akif/cookbook/ingredient/IngredientRepository.java
package dev.akif.cookbook.ingredient;

import org.springframework.data.repository.ListCrudRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Set;

@Repository
public interface IngredientRepository extends ListCrudRepository<IngredientEntity, Long> {
    List<IngredientEntity> findAllByIdIn(Set<Long> ids);

    List<IngredientEntity> findAllByNameIn(Set<String> names);
}

File: src/main/kotlin/dev/akif/cookbook/ingredient/IngredientRepository.kt
package dev.akif.cookbook.ingredient

import kotlinx.coroutines.flow.Flow
import org.springframework.data.repository.kotlin.CoroutineCrudRepository
import org.springframework.stereotype.Repository

@Repository
interface IngredientRepository: CoroutineCrudRepository<IngredientEntity, Long> {
    fun findAllByIdIn(ids: Set<Long>): Flow<IngredientEntity>

    fun findAllByNameIn(names: Set<String>): Flow<IngredientEntity>
}
